graphix/pyzx.py

```diff
@@ -84,23 +87,30 @@ def add_vertices(n: int, ty: zx.VertexType) -> list[zx.VertexType]:
 
     og_edges = og.inside.edges()
     pyzx_edges = ((map_to_pyzx[a], map_to_pyzx[b]) for a, b in og_edges)
-    g.add_edges(pyzx_edges, zx.EdgeType.HADAMARD)
+    g.add_edges(pyzx_edges, EdgeType.HADAMARD)
 
     # Add the edges between the Z spiders in the graph body
     for og_index, meas in og.measurements.items():
         # If it's an X measured node, then we handle it in the next loop
         if meas.plane == Plane.XY:
-            g.set_phase(map_to_pyzx[og_index], meas.angle)
+            g.set_phase(map_to_pyzx[og_index], Fraction(meas.angle))
 
     # Connect the X measured vertices
     for og_index, pyzx_index in zip(x_meas, x_meas_verts):
-        g.add_edge((map_to_pyzx[og_index], pyzx_index), zx.EdgeType.HADAMARD)
-        g.set_phase(pyzx_index, og.measurements[og_index].angle)
+        g.add_edge((map_to_pyzx[og_index], pyzx_index), EdgeType.HADAMARD)
+        g.set_phase(pyzx_index, Fraction(og.measurements[og_index].angle))
 
     return g
 
 
-def from_pyzx_graph(g: BaseGraph) -> OpenGraph:
+def _checked_float(x: FractionLike) -> float:
+    if not isinstance(x, (Fraction, int)):
```

Comment:
Merged in ebdea88cfc8b426d3f6ccf2829c30d957845a7b2 .

Let me use `FractionLike` for `x` as this can be `Poly` instance that cannot be converted to `float`.